app/src/main/java/app/gamenative/utils/OpenContainerScanner.kt

```
@@ -0,0 +1,460 @@
+        if (!uniqueExeRel.isNullOrEmpty()) {
+            val exeFile = File(folder, uniqueExeRel.replace('/', File.separatorChar))
+            if (exeFile.exists()) {
+                val outIco = File(exeFile.parentFile, exeFile.nameWithoutExtension + ".extracted.ico")
+                // Use cache if up to date, else (re)extract
+                val useCached = outIco.exists() && outIco.lastModified() >= exeFile.lastModified()
+                if (useCached) return outIco.absolutePath
+                try {
+                    if (ExeIconExtractor.tryExtractMainIcon(exeFile, outIco)) {
+                        return outIco.absolutePath
+                    }
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)